src/poetry/console/commands/add.py

```diff
@@ -153,6 +162,11 @@ def handle(self) -> int:
 
         if not packages:
             self.line("Nothing to add.")
+
+            if packages_migrated and not self.option("dry-run"):
```

Comment:
I'm not sure about writing when we have nothing to add -- it seems more useful to only run when a change is actually made to me.